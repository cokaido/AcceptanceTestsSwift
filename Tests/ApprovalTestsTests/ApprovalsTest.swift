import XCTest
//#if os(OSX)
//@testable import ApprovalTests_Swift
//#elseif os(iOS)
//@testable import ApprovalTests_iOS
//#endif
@testable import ApprovalTests

class ApprovalsTest: XCTestCase {
    func testToString() throws {
        try Approvals.verify(CGRect(x: 5, y: 10, width: 100, height: 200))
    }

    func testAsJson() throws {
        try Approvals.verifyAsJson(CGRect(x: 5, y: 10, width: 100, height: 200))
    }

    func testAsJsonWithCustomObject() throws {
        try Approvals.verifyAsJson(MyRect(x: 5, y: 10, width: 100, height: 200))
    }
}

class MyRect: Codable {
    var x: Int
    var y: Int
    var width: Int
    var height: Int

    init(x: Int, y: Int, width: Int, height: Int) {
        self.x = x
        self.y = y
        self.width = width
        self.height = height
    }
}
