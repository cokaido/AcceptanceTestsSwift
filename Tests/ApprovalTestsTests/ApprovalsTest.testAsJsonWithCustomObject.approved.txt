{
  "y" : 10,
  "x" : 5,
  "width" : 100,
  "height" : 200
}